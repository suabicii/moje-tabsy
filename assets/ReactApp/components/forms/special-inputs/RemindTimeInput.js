import React from "react";

function RemindTimeInput({handleUnitInputChange, ordinalNumber}) {
    return (
        <div className="input-group mt-3">
            <input type="number" name={`remindTime${ordinalNumber}`} id={`remindTime${ordinalNumber}`}
                   className="form-control" min="1"
                   data-testid="remindTime" required/>
            <select name={`unit${ordinalNumber}`} id={`unit${ordinalNumber}`} className="form-select text-center" defaultValue="min"
                    onChange={handleUnitInputChange}>
                <option value="min">min.</option>
                <option value="h">h</option>
                <option value="atDosingHour">W godzinie za≈ºycia</option>
            </select>
            <span className="input-group-text ps-3" id={`remindTimeSuffix${ordinalNumber}`}>przed</span>
        </div>
    );
}

export default RemindTimeInput;