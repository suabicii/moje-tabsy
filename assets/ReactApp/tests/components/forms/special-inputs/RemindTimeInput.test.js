import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import RemindTimeInput from "../../../../components/forms/special-inputs/RemindTimeInput";

it('should correctly render remind time input', () => {
    const renderer = new ReactShallowRenderer();
    const handleUnitInputChange = jest.fn();
    renderer.render(<RemindTimeInput handleUnitInputChange={handleUnitInputChange} ordinalNumber={1}/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});