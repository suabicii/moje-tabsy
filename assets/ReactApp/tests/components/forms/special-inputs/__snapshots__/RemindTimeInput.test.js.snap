// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render remind time input 1`] = `
<div
  className="input-group mt-3"
>
  <input
    className="form-control"
    data-testid="remindTime"
    id="remindTime1"
    min="1"
    name="remindTime1"
    required={true}
    type="number"
  />
  <select
    className="form-select text-center"
    defaultValue="min"
    id="unit1"
    name="unit1"
    onChange={[MockFunction]}
  >
    <option
      value="min"
    >
      min.
    </option>
    <option
      value="h"
    >
      h
    </option>
    <option
      value="atDosingHour"
    >
      W godzinie za≈ºycia
    </option>
  </select>
  <span
    className="input-group-text ps-3"
    id="remindTimeSuffix1"
  >
    przed
  </span>
</div>
`;
